server/controllers/answers.js

```diff
@@ -1,83 +1,91 @@
+        res.status(400).json({ status: 'Error', msg: 'question_id is required and must be a number' });
+      } else {
+        models.answers.query(questionId, count, page)
+          .then((response) => {
+            const { rows: results } = response;
+            res.status(200).json({
+              status: 'OK',
+              data: {
+                question: questionId, page, count, results,
+              },
+            });
+          })
+          .catch(next);
+      }
+    } catch {
+      res.status(500).json({ msg: 'Internal database error gettings answers' });
+      next();
     }
   },
-  post: (req, res) => {
-    const { question_id } = req.params;
-    const questionId = question_id.toString();
-    const {
-      body, name, email, photos,
-    } = req.body;
-    if (!body || typeof body !== 'string' || body.length > 1000) {
-      res.status(400).json({ status: 'Error', msg: 'answer body must be string <= 1000 chars in length' });
-    } else if (!name || typeof name !== 'string' || name.length > 60) {
-      res.status(400).json({ status: 'Error', msg: 'name must be string <= 60 chars in length' });
-    } else if (!email || typeof email !== 'string' || email.length > 60) {
-      // TODO: use regex for e-mail verification
-      res.status(400).json({ status: 'Error', msg: 'email must be string <= 60 chars in length' });
-    } else if (!Array.isArray(photos) || photos.some((photo) => typeof photo !== 'string')) {
-      res.status(400).json({ status: 'Error', msg: 'photos must be array of strings' });
-    } else {
-      models.answers.create(questionId, body, name, email, photos)
-        .then((response) => {
-          res.status(201).json({ status: 'CREATED' });
-          // TODO: handle duplicate entry gracefully
-        })
-        .catch((err) => {
-          console.error('Internal database error posting answer\n', err);
-          res.status(500).json({ msg: 'Internal database error posting answer\n' });
-        });
+  post: (req, res, next) => {
+    try {
+      const { question_id } = req.params;
+      const questionId = question_id.toString();
+      const {
+        body, name, email, photos,
+      } = req.body;
+      if (!body || typeof body !== 'string' || body.length > 1000) {
+        res.status(400).json({ status: 'Error', msg: 'answer body must be string <= 1000 chars in length' });
+      } else if (!name || typeof name !== 'string' || name.length > 60) {
+        res.status(400).json({ status: 'Error', msg: 'name must be string <= 60 chars in length' });
+      } else if (!email || typeof email !== 'string' || email.length > 60) {
+        // TODO: use regex for e-mail verification
+        res.status(400).json({ status: 'Error', msg: 'email must be string <= 60 chars in length' });
+      } else if (!Array.isArray(photos) || photos.some((photo) => typeof photo !== 'string')) {
+        res.status(400).json({ status: 'Error', msg: 'photos must be array of strings' });
+      } else {
```

Comment:
I'm not sure when an Array of strings is sent over a HTTP request it will come through as a JS array. I think it may come as a stringified array that you'll have to parse.

Reply:
Oh, that's a good point. I just ran a test w/ postman and I think params (i.e. /:product_id) and query params (i.e. ?page=5&count=5) are stringified, but POST body is parsed back into a JSON after it's stringified (to match HTTP protocol).